src/Disks/ObjectStorages/Local/LocalObjectStorage.cpp

```diff
@@ -40,47 +40,12 @@ bool LocalObjectStorage::exists(const StoredObject & object) const
     return fs::exists(object.remote_path);
 }
 
-std::unique_ptr<ReadBufferFromFileBase> LocalObjectStorage::readObjects( /// NOLINT
-    const StoredObjects & objects,
-    const ReadSettings & read_settings,
-    std::optional<size_t>,
-    std::optional<size_t>) const
-{
-    auto modified_settings = patchSettings(read_settings);
-    auto global_context = Context::getGlobalContextInstance();
-    auto read_buffer_creator = [=](bool /* restricted_seek */, const StoredObject & object) -> std::unique_ptr<ReadBufferFromFileBase>
-    { return std::make_unique<ReadBufferFromFile>(object.remote_path); };
-
-    return std::make_unique<ReadBufferFromRemoteFSGather>(
-        std::move(read_buffer_creator),
-        objects,
-        "file:",
-        modified_settings,
-        global_context->getFilesystemCacheLog(),
-        /* use_external_buffer */ false);
-}
-
 ReadSettings LocalObjectStorage::patchSettings(const ReadSettings & read_settings) const
 {
-    if (!read_settings.enable_filesystem_cache)
-        return IObjectStorage::patchSettings(read_settings);
-
     auto modified_settings{read_settings};
-    /// For now we cannot allow asynchronous reader from local filesystem when CachedObjectStorage is used.
-    switch (modified_settings.local_fs_method)
-    {
-        case LocalFSReadMethod::pread_threadpool:
-        case LocalFSReadMethod::pread_fake_async:
-        {
-            modified_settings.local_fs_method = LocalFSReadMethod::pread;
-            LOG_INFO(log, "Changing local filesystem read method to `pread`");
-            break;
-        }
-        default:
-        {
-            break;
-        }
-    }
+    /// Other options might break assertions in AsynchronousBoundedReadBuffer.
+    modified_settings.local_fs_method = LocalFSReadMethod::pread;
+    modified_settings.direct_io_threshold = 0; /// Disable.
```

Comment:
you can do this change as part of a new PR but please add some kind of explanation
just writing `Disable` doesn't seem that useful

Reply:
In general the comment is the same as for update of `local_fs_method` one line above, e.g. `/// Other options might break assertions in AsynchronousBoundedReadBuffer.`. There is an assertion which used to fail before in master as well, but we fixed it by just removing `AsynchronousBoundedReadBuffer` from `LocalObjectStorage`. In this PR I returned it back, so needed another way to fix failing assertion, namely this one https://github.com/ClickHouse/ClickHouse/blob/61b6a93ab3b37b004231da05c304c6c9699b05f1/src/Disks/IO/ThreadPoolRemoteFSReader.cpp#L155 e.g. those settings lead to using different buffer implementations which do not use buffer in an expected way. Probably there should be another way to fix this instead of disabling, but as `LocalObjectStorage` is a feature for testing purposes convenience, it does not matter to support all read methods, any one is sufficient.